pkg/alertmanager/alerts/local/store.go

```diff
@@ -0,0 +1,79 @@
+package local
+
+import (
+	"context"
+	"flag"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/prometheus/alertmanager/config"
+
+	"github.com/cortexproject/cortex/pkg/alertmanager/alerts"
+)
+
+// StoreConfig configures a static file alertmanager store
+type StoreConfig struct {
+	Path string `yaml:"path"`
+}
+
+// RegisterFlags registers flags related to the alertmanager file store
+func (cfg *StoreConfig) RegisterFlags(f *flag.FlagSet) {
+	f.StringVar(&cfg.Path, "alertmanager.storage.local.path", "", "Path at which alertmanager configurations are stored.")
+}
+
+// Store is used to load user alertmanager configs from a local disk
+type Store struct {
+	cfg StoreConfig
+}
+
+// NewStore returns a new file alert store.
+func NewStore(cfg StoreConfig) (*Store, error) {
+	return &Store{cfg}, nil
+}
+
+// ListAlertConfigs returns a list of each users alertmanager config.
+func (f *Store) ListAlertConfigs(ctx context.Context) (map[string]alerts.AlertConfigDesc, error) {
+	configs := map[string]alerts.AlertConfigDesc{}
+	err := filepath.Walk(f.cfg.Path, func(path string, info os.FileInfo, err error) error {
```

Comment:
In this callback function, whenever we do `return err` I would suggest to wrap the error to add more details about the failed operation. Could be easier to investigate issues, cause filesystem-related errors are usually quite generic.

Reply:
Will do.